docs/doxygen/demos/jobs_demo.dox

```diff
@@ -0,0 +1,68 @@
+/**
+@page jobs_demo AWS IoT Jobs
+
+This demonstration downloads files from URLs received via AWS IoT Jobs.
+(See https://docs.aws.amazon.com/iot/latest/developerguide/iot-jobs.html for an introduction.)
+
+This demo is intended for Linux platforms with the GCC toolchain, curl,
+and libmosquitto installed.  To build this demo, run make.
```

Comment:
To avoid ambiguity about the location of running `make` for the reader.
```suggestion
and libmosquitto installed.  To build this demo, run make in the `demos/jobs/jobs_demo_mosquitto` folder.
```